modules/internal/ChapelRange.chpl

```diff
@@ -2456,9 +2461,11 @@ proc _cast(type t, r: range(?)) where isRangeType(t) {
 
     modulus = abs(modulus);
     var m = modulus : minType;
-    if minType != modulus.type {
-      if m : modulus.type != modulus then
-        halt("Modulus too large.");
+    if boundsChecking {
+      if minType != modulus.type {
+        if m : modulus.type != modulus then
+          HaltWrappers.boundsCheckHalt("Modulus too large.");
```

Comment:
Doesn't really feel like bounds checking either. Is this more of a casting error?